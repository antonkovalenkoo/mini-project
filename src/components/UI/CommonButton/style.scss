.CommonButton {
  width: 250px;
  height: 40px;
  border-radius: 10px;
  font-size: 15px;
  font-weight: 700;
  border: none;
  background-color: $button-color-primary;
  color: white;
  transition: all 0.3s;

  &:hover {
    transform: scale(1.01);
    box-shadow: rgba(50, 50, 93, 0.25) 0px 13px 27px -5px, rgba(0, 0, 0, 0.3) 0px 8px 16px -8px;
  }
}