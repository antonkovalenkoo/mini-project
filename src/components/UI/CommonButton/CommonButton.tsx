import { ButtonHTMLAttributes, FC } from 'react';
import './style.scss';

const CommonButton: FC<ButtonHTMLAttributes<HTMLButtonElement>> = ({
  ...props
}) => {
  return (
    <button {...props} className='CommonButton' />
  )
}

export default CommonButton;
