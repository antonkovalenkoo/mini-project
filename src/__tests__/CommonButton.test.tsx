import '@testing-library/jest-dom';
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import CommonButton from "../components/UI/CommonButton/CommonButton";

const props = {
  type: 'submit' as 'submit'
}

describe('CommonButton component test', () => {
  test('CommonButton renders without props', () => {
    render(<CommonButton />);

    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  test('CommonButton onClick working', () => {
    const onClickHandler = jest.fn();

    render(
      <CommonButton
        onClick={onClickHandler}
      />
    );

    userEvent.click(screen.getByRole('button'));
    expect(onClickHandler).toBeCalled();
  });

  test('CommonButton pass all button props to button', () => {
    render(
      <CommonButton {...props}/>
    );

    const buttonElement = screen.getByRole('button');
    expect(buttonElement).toMatchSnapshot();

    const passingProps = Object.entries(props);    
    passingProps.forEach(([ key, value ]) => {
      expect(buttonElement.getAttribute(key)).toEqual(value);
    });
  });
})