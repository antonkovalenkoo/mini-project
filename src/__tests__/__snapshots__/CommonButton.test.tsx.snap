// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonButton component test CommonButton pass all button props to button 1`] = `
<button
  class="CommonButton"
  type="submit"
/>
`;
